test/integration/test_expid.py

```diff
@@ -260,3 +279,337 @@ def test_copy_expid_no(autosubmit: Autosubmit) -> None:
+        "New Evaluation Experiment With Empty Current"])
+def test_create_new_experiment(mocker, test, operational, evaluation, expected, new_exp):
+    db_common_mock = mocker.patch('autosubmit.experiment.experiment_common.db_common')
+    if new_exp:
+        current_experiment_id = expected
+    else:
+        current_experiment_id = "empty"
+    _build_db_mock(current_experiment_id, db_common_mock, mocker)
+    experiment_id = new_experiment(_DESCRIPTION, _VERSION, test, operational, evaluation)
+    assert expected == experiment_id
+
+
+def test_copy_experiment_new(mocker):
+    db_common_mock = mocker.patch('autosubmit.experiment.experiment_common.db_common')
+    current_experiment_id = "empty"
+    _build_db_mock(current_experiment_id, db_common_mock, mocker)
+    experiment_id = copy_experiment(current_experiment_id, _DESCRIPTION, _VERSION, False, False, True)
+    assert "" == experiment_id
+
+
+def _build_db_mock(current_experiment_id, mock_db_common, mocker):
+    mock_db_common.last_name_used = mocker.Mock(return_value=current_experiment_id)
+    mock_db_common.check_experiment_exists = mocker.Mock(return_value=False)
+
+
+def test_autosubmit_generate_config(mocker, autosubmit):
+    expid = 'ff99'
+    original_local_root_dir = BasicConfig.LOCAL_ROOT_DIR
+    read_files_mock = mocker.patch('autosubmit.autosubmit.read_files')
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_file_path = os.path.join(temp_dir, 'files')
+        os.makedirs(temp_file_path)
+
+        with tempfile.NamedTemporaryFile(dir=temp_file_path, suffix='.yaml', mode='w') as source_yaml:
+            # Our processed and commented YAML output file must be written here
+            Path(temp_dir, expid, 'conf').mkdir(parents=True)
+            BasicConfig.LOCAL_ROOT_DIR = temp_dir
+
+            source_yaml.write(
+                dedent('''
+                    JOB:
+                        JOBNAME: SIM
+                        PLATFORM: local
+                    CONFIG:
+                        TEST: The answer?
+                        ROOT: No'''))
+            source_yaml.flush()
+            read_files_mock.return_value = Path(temp_dir)
+
+            parameters = {
+                'JOB': {
+                    'JOBNAME': 'sim'
+                },
+                'CONFIG': {
+                    'CONFIG.TEST': '42'
+                }
+            }
+            autosubmit.generate_as_config(exp_id=expid, parameters=parameters)
+
+            source_text = Path(source_yaml.name).read_text()
+            source_name = Path(source_yaml.name)
+            output_text = Path(temp_dir, expid, 'conf', f'{source_name.stem}_{expid}.yml').read_text()
+
+            assert source_text != output_text
+            assert '# sim' not in source_text
+            assert '# sim' in output_text
+            assert '# 42' not in source_text
+            print(output_text)
```

Comment:
We can remove all `print` statements. Pytest captures all the output and prints only when there are errors. But if you search for something like bad practice print tests you should find a few articles pointing why it's bad to rely on the text output during tests (they may lie to you, or just clutter the output and hide something useful from a traceback or from the test harness output).